kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/model/networking/NetworkingPlanner.java

```diff
@@ -118,11 +137,18 @@ private static ClusterIngressNetworkingDefinition clusterIngressNetworkingDefini
                                                                                          List<NodeIdRanges> nodeIdRanges,
                                                                                          @Nullable Tls tls) {
         ClusterIP clusterIP = ingress.getSpec().getClusterIP();
+        LoadBalancer loadBalancer = ingress.getSpec().getLoadBalancer();
         if (clusterIP != null) {
             return new ClusterIPClusterIngressNetworkingDefinition(ingress, cluster, primary, nodeIdRanges, tls);
         }
+        else if (loadBalancer != null) {
+            if (tls == null) {
+                throw new IngressPlanningException("Load balancer ingress requires downstream TLS to be provided by the virtualkafkacluster");
+            }
+            return new LoadBalancerClusterIngressNetworkingDefinition(ingress, cluster, loadBalancer, tls);
```

Comment:
Should the `LoadBalancerClusterIngressNetworkingDefinition` do the defensive thing and validate TLS is not null? Rather than protecting it externally?

Reply:
note to self: there should be a condition reporting if the user forgets to configure tls on loadBalancer. do we have it?

Reply:
there is a condition reported on the VKC complaining about the mismatch (using the existing logic from clusterIP).

Reply:
it's a bit funky, the parameter is NotNull, so I get warnings in IDEA if we don't have the guard before construction.  Also feels a bit weird to annotate it Nullable but fail immediately if it's null.